scripts/ImportExportTest.sh

```diff
@@ -0,0 +1,302 @@
+                                        return 0
+                                    fi
+                                done
+                            done
+
+                            # direct export via --asm-json, if imported with --import-asm-json.
+                            if ! $SOLC --asm-json --import-asm-json expected.asm | tail -n+4 > obtained_direct_import_export.json 2> obtained_direct_import_export.error
+                            then
+                                printf "\n"
+                                echo "$nth_input_file"
+                                cat obtained_direct_import_export.error
+                                FAILED=$((FAILED + 1))
+                                return 0
+                            else
+                                for obtained_contract in $(jq '.contracts | keys | .[]' obtained_direct_import_export.json 2> /dev/null)
+                                do
+                                    jq --raw-output ".contracts.${obtained_contract}.\"asm\"" obtained_direct_import_export.json > obtained_direct_import_export.asm
+                                    set +e
+                                    DIFF="$(diff expected.asm obtained_direct_import_export.asm)"
+                                    set -e
+                                    if [ "$DIFF" != "" ]
+                                    then
+                                        if [ "$DIFFVIEW" == "" ]
+                                        then
+                                            echo -e "ERROR: JSONS differ for $1: \n $DIFF \n"
+                                            echo "Expected:"
+                                            cat  expected.asm
+                                            echo "Obtained:"
+                                            cat obtained_direct_import_export.asm
+                                        else
+                                            # Use user supplied diff view binary
+                                            $DIFFVIEW expected.asm obtained_direct_import_export.asm
+                                        fi
+                                        _TESTED=
+                                        FAILED=$((FAILED + 1))
+                                        return 0
+                                    fi
+                                done
+                            fi
+
+                            rm obtained.json
+                            rm -f obtained.error
+                            for type in "${types[@]}"
+                            do
+                                rm "obtained.${type}"
+                            done
+                        fi
+
+                        for type in "${types[@]}"
+                        do
+                            rm "expected.${type}"
+                        done
+                    fi
+                done
+                rm expected.json
```

Comment:
I think there's no point removing these files given that the function runs in a loop that creates a new temporary dir in each cycle. So it will be gone anyway and for debugging you might want to inspect the files and it's easier to comment-out the single line deleting the whole dir than all the lines removing individual files.

Reply:
You're right!